### PR TITLE
[issue-2908]: Bump lake-builder version to 0.0.8

### LINKED ISSUE #2908: [Refactor][Build] Uplift Go version in Github Action
## What and why to refactor
Github unit-test action is building and running the tests with a pre 1.18 go version, so we can't run any 1.18+ syntax there at the moment.

## Describe the solution you'd like
Configure the action to use >= 1.18. (might as well use >= 1.19). Also consider using the "go-version-file" flag.

## Related issues
n/a

## Additional context
n/a

### PR DESCRIPTION
**Pull request by keon94**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [ ] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

# Summary

### Does this close any open issues?
Closes #2908 
Blocks #2940/#2984
Blocks #2907/#2941

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -41,5 +41,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.43.0
```
Comment (keon94) [inline .github/workflows/golangci-lint.yml:44]: 1.43 won't work with Go 1.18+, so had to upgrade

File: devops/lake-builder/Dockerfile [hunk 2]
```diff
@@ -14,20 +14,24 @@
...
 
-FROM golang:1.19.0-alpine3.16
+FROM golang:1.19.1-alpine3.16
 # libs/headers needed to compile gitextractor
+# copy headers over directly (it'll be a new dir)
 COPY --from=builder /usr/include /usr/include
-COPY --from=builder /usr/lib/libgit2.so /usr/lib/libgit2.so
-COPY --from=builder /usr/lib/libgit2.so.1.3 /usr/lib/libgit2.so.1.3
-COPY --from=builder /usr/lib/libpkgconf.so.3 /usr/lib/libpkgconf.so.3
+# copy all contents of the base lib to a temp directory to avoid overwrites of newer versions
```
Comment (keon94) [inline devops/lake-builder/Dockerfile:27]: safer, more robust solution

**Comment by keon94** [comment]
The linter errors must be due to the Linter version upgrade.

**Comment by keon94** [comment]
@klesh this is ready for review

**Review by mindlesscloud — APPROVED** [review]
LGTM

**Pull request merged**